File: Reduce Array Size to The Half.py
class Solution:
    def minSetSize(self, arr: List[int]) -> int:
        n = len(arr)
        mid = (n // 2) + (n % 2)
        counter = list(Counter(arr).values())
        counter.sort(reverse=True)
        for i, num in enumerate(counter):
            n -= num
            if n <= mid:
                return i+1
        return 0

File: Reduce Array Size to The Half.go
func minSetSize(arr []int) int {
    n := len(arr)
    mid := n / 2 + (n % 2)
    counter := make(map[int]int)
    for _, num := range arr {
        if _, ok := counter[num]; !ok {
            counter[num] = 1
        } else {
            counter[num]++
        }
    }

    vals := make([]int, 0)
    for _, num := range counter {
        vals = append(vals, -num)
    }
    sort.Ints(vals)
    for i, val := range vals {
        n += val
        if n <= mid {
            return i+1
        }
    }
    return 0
}
